pkg/informer/informer.go

```diff
@@ -58,14 +61,28 @@ const (
 	byGroupVersionResourceIndex = "byGroupVersionResource"
 )
 
-// DynamicDiscoverySharedInformerFactory is a SharedInformerFactory that
+// GVRSource is a source fo information about available GVRs,
+// and provides a way to register for changes in the available GVRs (addition of removal)
+// so that the list of informers can be updated.
+type GVRSource interface {
+	ListGVRs() []schema.GroupVersionResource
+	GetGVRPartialMetadata(gvr schema.GroupVersionResource) (*GVRPartialMetadata, error)
+	Synced() bool
+	Register(notifyUpdateNeeded func())
```

Comment:
Can we make this `Changes() <-chan whatever` - callback nesting is usually unpleasant to deal with :)

Reply:
I don't see how it would allow to implement what is [done in the `crdGVRSource`](https://github.com/kcp-dev/kcp/pull/2440/files#diff-e35b562e46d045590564268b506fe50d11eb737d43679891bac1238dd5c48762R964).
But maybe I'm simply not used enough to working with channels.

Reply:
Do you want me to provide you with an example, or are you going to poke around with an implementation on your own

Reply:
This should be good finally. Thank you.

Reply:
I'm preparing to push fixes to this PR, but first checking it in the context of my Proof PR (with the refactored Syncer using it)

Reply:
I implemented it this way: https://github.com/kcp-dev/kcp/blob/2ea4239c59b359f7262da33eb05dc0d064e6c230/pkg/informer/informer.go#L162-L172
And the `Changes()` channel, in the case of CRDs is fed this way:   https://github.com/kcp-dev/kcp/blob/2ea4239c59b359f7262da33eb05dc0d064e6c230/pkg/informer/informer.go#L922-L945